src/Core/Utility/ResultEx.re

```diff
@@ -8,6 +8,14 @@ let flatMap = f =>
   | Ok(x) => f(x)
   | Error(_) as err => err;
 
+let map2 = (f, a, b) => {
+  switch (a, b) {
+  | (Ok(vA), Ok(vB)) => Ok(f(vA, vB))
+  | (Error(_) as errA, _) => errA
+  | (_, Error(_) as errB) => errB
+  };
+};
+
```

Comment:
When it gets so complicated that you need to use `map2` it might be better to use the let ppx?

Reply:
Good idea, I forgot about that - I'll give it a try